source-mysql/discovery.go

```diff
@@ -305,6 +305,10 @@ func (db *mysqlDatabase) translateRecordField(isBackfill bool, columnType interf
 				}
 				return acc, nil
 			case "binary", "varbinary":
+				// Unlike BINARY(N) columns, VARBINARY(N) doesn't play around with adding or removing
```

Comment:
I'm kind of fuzzy on how the value resolution is all plumbed together, but am a little confused by the comment about a fallback for metadata predating the change to include the length for binary columns. The metadata gets requested anew every time the connector starts up right? Or maybe there's some other persistence mechanism I'm forgetting about right now?

Reply:
Replication decoding metadata is persisted in the state checkpoint between runs. It's originally initialized with the type information from discovery* and then thereafter it's updated only by DDL queries. So the concern here is metadata which predates this change and so doesn't have the binary type with length information, and won't unless/until it's either reinitialized by a backfill or altered in a DDL statement.

* Strictly speaking it is incorrect to do this because there's a data race if a table is altered right around the same time. But there is literally no alternative and in practice it works pretty reliably because people generally aren't altering their tables during that short window, and it only happens once (per binding and backfill counter increment).